cmake/HPX_SetupBoost.cmake

```diff
@@ -138,7 +140,7 @@ endif()
 
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 link_directories(${Boost_LIBRARY_DIRS})
-if(NOT MSVC)
+if((NOT MSVC) OR HPX_WITH_VCPKG)
```

Comment:
It looks to me that the variable to look out for is `HPX_WITH_BOOST_ALL_DYNAMIC_LINK` which requires to pass the libraries explcitly, right? That is, we need the libraries if we are not on MSVC or have `HPX_WITH_BOOST_ALL_DYNAMIC_LINK` set to `On`, right?
So if `HPX_WITH_VPKG` is requested, `HPX_WITH_BOOST_ALL_DYNAMIC_LINK` should default to `On`, I guess.